app/policies/service_provider_policy.rb

```diff
@@ -44,7 +44,11 @@ def admin?
     current_user.admin?
   end
 
-  def owner_or_admin?
-    owner? || admin?
+  def member?
+    sp.user_group.present? && sp.user_group == current_user.user_group
```

Comment:
I think we want to let them edit ones that they created too, right? like what if I create a service provider but am somehow not in the user group I assign to it?